modules/imgproc/src/shapedescr.cpp

```diff
@@ -342,7 +342,9 @@ namespace cv
 
 static inline Point2f getOfs(int i, float eps)
 {
-    return Point2f(((i & 1)*2 - 1)*eps, ((i & 2) - 1)*eps);
+    std::mt19937 gen(i);
+    std::uniform_real_distribution<float> dis(-eps, eps);
+    return Point2f(dis(gen), dis(gen));
```

Comment:
I propose to use cv::RNG for it to make it manageable outside:
- User may set seed to get deterministic behaviour
- OpenCV test system fixes cv::RNG seed for each test independently.

Reply:
Used `cv::RNG`